openverse_catalog/dags/providers/provider_api_scripts/freesound.py

```diff
@@ -0,0 +1,266 @@
+"""
+Content Provider:       Freesound
+
+ETL Process:            Use the API to identify all CC-licensed images.
+
+Output:                 TSV file containing the image, the respective
+                        meta-data.
+
+Notes:                  https://freesound.org/apiv2/search/text'
+                        No rate limit specified.
+"""
+import functools
+import logging
+import os
+
+import requests
+from common.licenses.licenses import get_license_info
+from common.loader import provider_details as prov
+from common.requester import DelayedRequester
+from common.storage.audio import AudioStore
+
+
+LIMIT = 150
+DELAY = 1  # in seconds
+RETRIES = 3
+HOST = "freesound.org"
+ENDPOINT = f"https://{HOST}/apiv2/search/text"
+PROVIDER = prov.FREESOUND_DEFAULT_PROVIDER
+# Freesound only has 'sounds'
+FREESOUND_CATEGORY = "sound"
+API_KEY = os.getenv("FREESOUND_API_KEY", "not_set")
+
+HEADERS = {
+    "Accept": "application/json",
+}
+DEFAULT_QUERY_PARAMS = {
+    "format": "json",
+    "token": API_KEY,
+    "query": "",
+    "page_size": LIMIT,
+    "fields": "id,url,name,tags,description,created,license,type,download,"
+    "filesize,bitrate,bitdepth,duration,samplerate,pack,username,"
+    "num_downloads,avg_rating,num_ratings,geotag,previews",
+}
+
+delayed_requester = DelayedRequester(DELAY)
+audio_store = AudioStore(provider=PROVIDER)
+
+logging.basicConfig(
+    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s",
+    level=logging.INFO,
+)
+logger = logging.getLogger(__name__)
+
+preview_bitrates = {
+    "preview-hq-mp3": 128000,
+    "preview-lq-mp3": 64000,
+    "preview-hq-ogg": 192000,
+    "preview-lq-ogg": 80000,
+}
+
+
+def main():
+    """This script pulls the data for a given date from the Freesound,
+    and writes it into a .TSV file to be eventually read
+    into our DB.
+    """
```

Comment:
As @AetherUnbound commented before, it doesn't use a date param so it actually pulls all the data. Better to update to avoid confusion.

Reply:
I've added the date :)